compiler/src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -14,8 +14,8 @@ import Symbols._
 import Flags.Module
 import reporting.{ThrowingReporter, Profile, Message}
 import collection.mutable
-import scala.concurrent.{Future, Await, ExecutionContext}
-import scala.concurrent.duration.Duration
+import util.concurrent.{Executor, Future}
```

Comment:
Wouldn't using `ExecutionContext.fromExecutor(Executors.newFixedThreadPool(1))`achieve the same result? What do we gain with custom futures?

Reply:
It's unclear. One thing we gain is we drop a bug dependency which might be useful when we move to Scala Native. Another is it's unclear what overhead the general scheduler would cause. The current solution is very small, so also easy to understand in its performance characteristics.